cfgov/jinja2/v1/_includes/molecules/notification.html

```diff
@@ -30,26 +31,18 @@
 
 {% macro render(type, is_visible, message, explanation=none, links=[]) %}
 
+  {# Lookup table for the CSS class and icon for each notification type. #}
   {% set type_lookup = {
-        'default':    'information-round',
-        'success':    'approved-round',
-        'warning':    'warning-round',
-        'error':      'error-round',
-        'money-tips': 'dollar-round',
-        'caution':   'information-round'
+        'information': ['', 'information-round'],
+        'success':     [' m-notification__success', 'approved-round'],
+        'warning':     [' m-notification__warning', 'warning-round'],
+        'error':       [' m-notification__error', 'error-round'],
+        'money-tips':  [' m-notification__success', 'dollar-round'],
+        'caution':     [' m-notification__error', 'information-round']
       }
   %}
-  {% set icon = type_lookup[type | default( 'default' )] %}
-
-  {% if type == 'money-tips' %}
-    {% set type = ' m-notification__success' %}
-  {% elif type == 'caution' %}
-    {% set type = ' m-notification__error' %}
-  {% elif type %}
-    {% set type = ' m-notification__' + type | string %}
-  {% else %}
-    {% set type = '' %}
-  {% endif %}
+  {% set icon = type_lookup[type | default( 'information' )][1] %}
+  {% set type = type_lookup[type | default( 'information' )][0] %}
```

Comment:
I wonder if we'd want to handle bad type values (like `default` would be now) by replacing them with `information` to prevent an error when accessing the lookup?

`set type = type if type in type_lookup else 'information'`

The downside is that passing in an erroneous value would essentially fail silently, but the upside is you'd get usable output instead of an error / broken page.

Reply:
I think we'd want the error. If someone mistyped `errorr` or whatever and ended up with an `information` notification, that would fail silently as you said.